### PR TITLE
k8splugin support manage_network_ns_lifecycle=true from crio

### PR DESCRIPTION
**Pull request by n3wt0nSAN**
#### Type of fix: 
Bug fix
#### Fixes #
[1054](https://github.com/contiv/netplugin/issues/1054)
Please describe:
- changes made in the Pull request
The cni 0.3 support the network namespace path can be `/proc/[pid]/ns/net` or a `bind-mount/link(like /var/run/netns/cni-<something>)` to it.
So, when the flag `manage_network_ns_lifecycle=true` on the **cri-o runtime**, **contiv** return the bug #1054.
I made some logical modifications to contiv run over buth net path. Some problems of permission on container, I can't execute the operation on the mounted path and that force me send part of the code to k8s_cni code.
Modifications: 
> On the function `addPod()` of file `driver.go` of the k8splugin  package
> And put the same logical on the function `addPodToContiv()`  of file` k8s_cni.go` of cniapi  package
- type of testing done (both manual and automated)
Manual tests on Kubernetes v1.10.5 with CRI-O v1.10.6 and Kata Container v1.1.0, and Kata needed the manage_network_ns_lifecycle=true, https://github.com/kata-containers/documentation/blob/master/architecture.md#cni. 

## TODO
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: mgmtfn/k8splugin/contivk8s/k8s_cni.go [hunk 1]
```diff
@@ -95,15 +279,40 @@ func addPodToContiv(nc *clients.NWClient, pInfo *cniapi.CNIPodAttr) {
 		}
 		os.Exit(1)
 	}
+	log.Infof("EP created IP: %s\n", result.Attr.IPAddress)
```
Comment (liucimin) [inline mgmtfn/k8splugin/contivk8s/k8s_cni.go]: Why change this function?
Why not change the 
// addPod is the handler for pod additions
func addPod(w http.ResponseWriter, r *http.Request, vars map[string]string) (interface{}, error) 

in \netplugin\mgmtfn\k8splugin\driver.go

Comment (n3wt0nSAN) [reply mgmtfn/k8splugin/contivk8s/k8s_cni.go]: When, the runtime send the pid process the operation inside the container is permitted, but when is the path of network namespace I have problems to exec same operation I get the error:  `RTNETLINK answers: Invalid argument, exit status 2`

`ls -lha /var/run/netns/`
```
-rw-r--r--  1 root root    0 Aug  9 18:41 cni-d5e070e5-08c4-46f0-96ee-4bb5e6c6e21a
lrwxrwxrwx  1 root root   55 Aug  9 18:41 k8s_contiv-blue-c1_default_01bc671d-9a69-11e8-a90e-000af70485d0_1-81a93eff -> /var/run/netns/cni-d5e070e5-08c4-46f0-96ee-4bb5e6c6e21a
```

So, the k8s_cni is part of the contivk8s plugin execute on the host. 

I make some test to tray maintain the logical on `./netplugin\mgmtfn\k8splugin\driver.go` but no success.

File: utils/ns/ns_linux.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+// WithNetNSPath executes the passed closure under the given network
+// namespace, restoring the original namespace afterwards.
+func WithNetNSPath(nspath string, toRun func(NetNS) error) error {
+	ns, err := GetNS(nspath)
+	if err != nil {
+		return err
+	}
+	defer ns.Close()
+	return ns.Do(toRun)
+}
```
Comment (liucimin) [inline utils/ns/ns_linux.go]: maybe these files can use vendor in the project?
https://github.com/containernetworking/plugins/tree/master/pkg/ns

**Comment by n3wt0nSAN** [comment]
Yes. Is good practice. But the master version of this vendor don't have the
function I need. Only
https://github.com/containernetworking/plugins/tree/v0.7/pkg/ns
<https://github.com/containernetworking/plugins/tree/master/pkg/ns>.

You now how I get this specific branch with *go get*?

TIA

**Pull request closed without merging**